mongo/change_stream.go

```diff
@@ -553,6 +573,23 @@ func (cs *ChangeStream) loopNext(ctx context.Context, nonBlocking bool) {
 	}
 }
 
+func (cs *ChangeStream) isResumableError() bool {
+	commandErr, ok := cs.err.(CommandError)
+	if !ok || commandErr.HasErrorLabel(networkErrorLabel) {
+		// All non-server errors or network errors are resumable.
+		return true
+	}
+
+	// For wire versions 9 and above, a server error is resumable if it has the ResumableChangeStreamError label.
+	if cs.wireVersion != nil && cs.wireVersion.Includes(minResumableLabelWireVersion) {
```

Comment:
Minor point, but could cs.wireVersion ever be nil at this point?

Reply:
It should not be, but I have it this way for consistency because this is a common check through the rest of our codebase.